Framework/DataHandling/src/LoadEmptyInstrument.cpp

```diff
@@ -98,15 +102,34 @@ void LoadEmptyInstrument::init() {
  *values
  */
 void LoadEmptyInstrument::exec() {
+  Progress prog(this, 0.0, 1.0, 10);
+
   // load the instrument into this workspace
+  MatrixWorkspace_sptr ws;
+  const std::string instrumentName = getPropertyValue("InstrumentName");
   const std::string filename = getPropertyValue("Filename");
-  const std::string instrumentname = getPropertyValue("InstrumentName");
-  Instrument_const_sptr instrument;
-  Progress prog(this, 0.0, 1.0, 10);
+  if (!instrumentName.empty())
+    ws = this->runLoadInstrument(filename, instrumentName);
+  else {
+    std::string ext{std::filesystem::path(filename).extension().string()};
+    FilenameExtension enFilenameExtension(ext);
+    switch (enFilenameExtension) {
+    case FilenameExtensionEnum::XML:
+    case FilenameExtensionEnum::HDF5:
+      ws = this->runLoadInstrument(filename, instrumentName);
+      break;
+    case FilenameExtensionEnum::NXS:
+      ws = this->runLoadInstrumentFromNexus(filename);
+      break;
+    default:
+      std::ostringstream os;
+      os << "Instrument file has an invalid extension: "
+         << "\"" << ext << "\"";
+      throw std::runtime_error(os.str());
```

Comment:
This is the primary functional change in the program.  Are there any new tests to support this new behavior?

Reply:
Added `testLoadFromNexus`, `testLoadFromNexusNoIDF` and `testLoadFileWithInvalidExtension` to `LoadEmptyInstrumentTest`.